integration/spark/shared/src/main/java/io/openlineage/spark/agent/lifecycle/plan/KustoRelationVisitor.java

```diff
@@ -0,0 +1,161 @@
+package io.openlineage.spark.agent.lifecycle.plan;
+
+import io.openlineage.client.OpenLineage;
+import io.openlineage.spark.api.DatasetFactory;
+import io.openlineage.spark.api.OpenLineageContext;
+import io.openlineage.spark.api.QueryPlanVisitor;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.reflect.FieldUtils;
+import org.apache.spark.api.java.Optional;
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan;
+import org.apache.spark.sql.execution.datasources.LogicalRelation;
+import org.apache.spark.sql.sources.BaseRelation;
+import org.apache.spark.sql.sources.CreatableRelationProvider;
+import org.apache.spark.sql.types.StructType;
+
+/** */
+@Slf4j
+public class KustoRelationVisitor<D extends OpenLineage.Dataset>
+    extends QueryPlanVisitor<LogicalRelation, D> {
+
+  private final DatasetFactory<D> factory;
+  private static final String KUSTO_CLASS_NAME =
+      "com.microsoft.kusto.spark.datasource.KustoRelation";
+
+  private static final String KUSTO_PROVIDER_CLASS_NAME =
+      "com.microsoft.kusto.spark.datasource.DefaultSource"; // Not used
+
+  private static final String KUSTO_URL_SUFFIX = ".kusto.windows.net";
+
+  private static final String KUSTO_PREFIX = "azurekusto://";
+
+  public KustoRelationVisitor(OpenLineageContext context, DatasetFactory<D> factory) {
+    super(context);
+    this.factory = factory;
+  }
+
+  public static boolean isKustoClass(LogicalPlan plan) {
+    return plan instanceof LogicalRelation
+        && ((LogicalRelation) plan).relation().getClass().getName().equals(KUSTO_CLASS_NAME);
+  }
+
+  public static boolean isKustoSource(CreatableRelationProvider provider) {
+    return provider.getClass().getName().equals(KUSTO_PROVIDER_CLASS_NAME);
+  }
+
+  public static boolean hasKustoClasses() {
+    try {
+      KustoRelationVisitor.class
+          .getClassLoader()
+          .loadClass("com.microsoft.kusto.spark.datasource.DefaultSource");
+      return true;
+    } catch (Exception e) {
+      // swallow - we don't care
+    }
+    try {
+      Thread.currentThread()
```

Comment:
We kept both just in case there is a setup we didn't consider in which the thread classloader misses the class, but the class.getClassLoader captures it, in which case we would still return true. But happy to switch it to the single check using Thread.currentThread.getClassLoader if you recommend that